### PR TITLE
Admin discount page update

### PR DESCRIPTION
**Pull request by davidVetter**
Should take care of most things left on admin discount page, added log out button to nav bar, validation on some forms, indicator for is discount is valid, expired or turned off, added two updates to alter statements on the database

### CODE REVIEW

File: server/routes/discounts.router.js [hunk 1]
```diff
@@ -139,6 +139,28 @@ router.put("/", rejectUnauthenticated, rejectUnauthorizedUser, (req, res) => {
     });
 }); // End edit discount PUT
 
+// This PUT will toggle the is_shown for a discount by id
+router.put("/active", rejectUnauthenticated, rejectUnauthorizedUser, (req, res) => {
```
Comment (Lloydrp) [inline server/routes/discounts.router.js:143]: I believe this should have rejectNonAdmin, not rejectUnauthorized.

File: server/routes/discounts.router.js [hunk 2]
```diff
@@ -139,6 +139,28 @@ router.put("/", rejectUnauthenticated, rejectUnauthorizedUser, (req, res) => {
...
+  console.log("In discount PUT with: ", req.body);
```
Comment (Lloydrp) [inline server/routes/discounts.router.js:144]: Comment Out Console Log, Please

File: src/components/AddDiscount/AddDiscount.jsx [hunk 3]
```diff
@@ -59,32 +85,77 @@ function AddDiscountModal() {
...
+  const handleSelectVendor = (eventKey) => {
+    setVendorId(eventKey);
+    setVendorSelected(true);
+    setVendorNotSelected(true);
+  }
+
+  const handleSelectCategory = (eventKey) => {
+    setCategoryId(eventKey);
+    setCategorySelected(true);
+    setVendorNotSelected(true);
```
Comment (Lloydrp) [inline src/components/AddDiscount/AddDiscount.jsx:110]: Should this be setCategoryNotSelected?

File: src/components/UserPage/UserPage.jsx [hunk 4]
```diff
@@ -76,7 +76,7 @@ function UserPage() {
             {allIconComponents.editUser} Edit Info
           </button>
           <div className="w-100 d-flex justify-content-center">
-          <LogOutButton className="btn btn-primary mt-2 col col-lg-6" />
+          <LogOutButton className="btn btn-primary mt-2 col col-md-9 col-lg-6" />
```
Comment (Lloydrp) [inline src/components/UserPage/UserPage.jsx:79]: If we are putting a logout in the navbar, we should likely just remove the logout button in the user page.  Makes it redundant when it's accessible on any page in he nav bar.

File: src/redux/sagas/discount.saga.js [hunk 5]
```diff
@@ -57,6 +57,17 @@ function* editDiscount(action) {
   }
 } // End edit discount
 
+// PUT to toggle discount is_shown by discount_id
+function* toggleActiveDiscount(action) {
+  try {
+    console.log("In toggle discount with: ", action.payload);
```
Comment (Lloydrp) [inline src/redux/sagas/discount.saga.js:63]: Comment out console log, please

**Review by Lloydrp — CHANGES_REQUESTED** [review]

File: src/components/UserPage/UserPage.jsx (hunk 4 above)
Comment (davidVetter) [reply src/components/UserPage/UserPage.jsx:79]: I agree. Removed

File: src/components/AddDiscount/AddDiscount.jsx (hunk 3 above)
Comment (davidVetter) [reply src/components/AddDiscount/AddDiscount.jsx:110]: Changed

File: server/routes/discounts.router.js (hunk 2 above)
Comment (davidVetter) [reply server/routes/discounts.router.js:144]: Removed

File: server/routes/discounts.router.js (hunk 1 above)
Comment (davidVetter) [reply server/routes/discounts.router.js:143]: Changed

File: src/redux/sagas/discount.saga.js (hunk 5 above)
Comment (davidVetter) [reply src/redux/sagas/discount.saga.js:63]: Removed

**Review by Lloydrp — APPROVED** [review]

**Pull request merged**